src/main/java/gregtech/client/model/modelfactories/OreBakedModel.java

```diff
@@ -21,73 +24,72 @@
 import net.minecraft.world.World;
 import net.minecraftforge.client.MinecraftForgeClient;
 import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.lang3.tuple.Pair;
-import org.lwjgl.util.vector.Vector3f;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import javax.vecmath.Matrix4f;
 import java.util.*;
 
+@SideOnly(Side.CLIENT)
 public class OreBakedModel implements IBakedModel {
 
     public static final OreBakedModel INSTANCE = new OreBakedModel();
 
-    private final Map<StoneType, IBakedModel> stoneTypeModels;
-    private final Map<MaterialIconSet, Map<EnumFacing, BakedQuad>> materialFaces;
+    private final Map<StoneType, TextureAtlasSprite> stoneTypeModels;
+    private final Table<StoneType, EnumFacing, BakedQuad> cacheBottom;
+    private final Table<MaterialIconSet, EnumFacing, BakedQuad[]> cacheTop;
     private final ThreadLocal<TextureAtlasSprite> particle;
+    private static IBakedModel model;
 
    private OreBakedModel() {
        this.stoneTypeModels = new Object2ObjectOpenHashMap<>();
-       this.materialFaces = new Object2ObjectOpenHashMap<>();
+       this.cacheBottom = Tables.newCustomTable(Maps.newHashMap(), () -> Maps.newEnumMap(EnumFacing.class));
+       this.cacheTop = Tables.newCustomTable(Maps.newHashMap(), () -> Maps.newEnumMap(EnumFacing.class));
```

Comment:
Very concerning caches that doesn't need to exist

Reply:
any suggestions

Reply:
There's just wasn't a need for caches here, not sure why we would need them now in this changed impl.

Reply:
due to the z-fighting of the ore and the stone layer in the previous code.